src/osp/Satellites/SatActiveArea.h

```diff
@@ -28,78 +28,53 @@
 
 #include <map>
 
-//#include <Magnum/Math/Color.h>
-//#include <Magnum/PixelFormat.h>
-//#include <Magnum/SceneGraph/Camera.h>
-
 #include "../types.h"
-//#include "../scene.h"
+#include "../Active/activetypes.h"
 
 #include "../Resource/Package.h"
 #include "../Resource/PrototypePart.h"
 
-//#include "../Active/ActiveScene.h"
-//#include "../OSPApplication.h"
-//#include "../UserInputHandler.h"
-//#include "../Active/FtrNewtonBody.h"
-
 namespace osp::universe
 {
 
 
 class OSPMagnum;
 class SatActiveArea;
 
+struct UCompActivatable { };
+
+struct UCompActivationMutable
+{
+    Satellite m_area{entt::null};
+    active::ActiveEnt m_ent{entt::null};
+};
+
+struct UCompActivationRadius
+{
+    float m_radius;
+};
 
 struct UCompActiveArea
 {
-    //active::ActiveEnt m_camera;
 
-    unsigned m_sceneIndex;
+    // for later use
+    //active::MapActiveScene_t::iterator m_scene;
 
     // true when the ActiveArea is moving
     bool m_inMotion;
 };
 
 
-class SatActiveArea : public CommonTypeSat<SatActiveArea, UCompActiveArea>
+class SatActiveArea : CommonTypeSat<SatActiveArea, UCompActiveArea>
```

Comment:
Private inheritance?

Reply:
Should this be explicit?

Reply:
Generally I prefer to explicitly say public / private for the inheritance because it changes the question people ask from

"Did you know this makes it private inheritance?"
to
"Could you explain why you want private inheritance?"

I don't object to private inheritance, just wasn't sure it's what you meant.